pgcontents/tests/test_synchronization.py

```diff
@@ -367,29 +339,24 @@ def check_call(kwargs, expect_checkpoints_content):
                 nb = result['content']
                 manager.mark_trusted_cells(nb, result['path'])
 
-                # Check that the checkpoint content matches what's expected
-                key = (result['user_id'], result['path'],
-                       result['last_modified'])
-                self.assertEqual(nb, expect_checkpoints_content[key])
-
-                checkpoint_record.append(key)
+                checkpoint_record.append((result['user_id'], result['path'],
+                                          result['last_modified'], nb))
 
-            # Make sure all checkpoints were found
-            self.assertEqual(sorted(checkpoint_record),
-                             sorted(expect_checkpoints))
+            # Make sure all checkpoints were found in the right order
+            self.assertEqual(checkpoint_record, expect_checkpoints)
 
         # No `min_dt`/`max_dt`
-        check_call({}, merge_dicts(beginning_content,
-                                   middle_content, end_content))
+        check_call({}, sum([beginning_checkpoints, middle_checkpoints,
```

Comment:
Using `sum` to concat lists, while nice and succinct, feels close to being "too cute".  (It's also O(N^2), though that doesn't really matter here).  One way to help clarify what's happening here might be to write a short helper function inline:

```python
def concat_all(lists):
    return sum(lists, [])
```

I don't feel super strongly about this one though.